### PR TITLE
Minor RC refactors part 3

### PR DESCRIPTION
**Pull request by tummychow**
intermediate PR for things that RU depends on, but are not themselves part of RU.

### CODE REVIEW

File: pkg/kp/rcstore/consul_store.go [hunk 1]
```diff
@@ -220,6 +220,16 @@ func (s *consulStore) SetDesiredReplicas(id fields.ID, n int) error {
 	})
 }
 
+func (s *consulStore) MutateDesiredReplicas(id fields.ID, n int) error {
```
Comment (mpuncel) [inline pkg/kp/rcstore/consul_store.go]: what about "Add" instead of "Mutate". I thought the second argument was the desired value rather than an operand

**Comment by mpuncel** [comment]
:+1:  just that one nit on function name

File: pkg/kp/rcstore/consul_store.go (hunk 1 above)
Comment (tummychow) [inline pkg/kp/rcstore/consul_store.go]: yeah that's a good change. anyone else have opinions on this name? otherwise I'll use AddDesiredReplicas instead

Comment (petertseng) [inline pkg/kp/rcstore/consul_store.go]: yup, I would prefer Add rather than Mutate

File: pkg/kp/rcstore/store.go [hunk 2]
```diff
@@ -33,7 +33,10 @@ type Store interface {
 	// session that deletes its keys on invalidation.
 	Lock(id fields.ID, session string) (bool, error)
 
+	// Set the desired replica count for the given RC to the given integer.
 	SetDesiredReplicas(fields.ID, int) error
```
Comment (petertseng) [inline pkg/kp/rcstore/store.go]: I take it that you feel that both `Set` and `Add` are still necessary and worth keeping?

**Comment by petertseng** [comment]
Also 👍

File: pkg/kp/rcstore/store.go (hunk 2 above)
Comment (tummychow) [inline pkg/kp/rcstore/store.go]: yes I think both are useful. Add happened to be an efficiency win for the code I'm writing at the moment, but I think Set is a more obvious interface for humans (rctl), and it's also the only way to clear an RC to zero in a single call.

Comment (petertseng) [inline pkg/kp/rcstore/store.go]: seems reasonable

**Pull request merged**